### PR TITLE
swaglog: enable type-checked against format string.

### PR DESCRIPTION
**Pull request by deanlee**
enable type-check & correct format string.

added a macro `SWAG_LOG_CHECK_FMT` to causes the compiler to check the arguments in calls to `cloudlog` for consistency with the `printf` style format string argument:
```

system/ubloxd/ublox_msg.cc:393:57: error: format specifies type 'int' but the argument has type 'uint64_t' (aka 'unsigned long') [-Werror,-Wformat]
      LOGE("SV_ID != SLOT_NUMBER: %d %d", msg->sv_id(), data->n());
                                     ~~                 ^~~~~~~~~
                                     %lu
```

### CODE REVIEW

**Comment by deanlee** [comment]
failed to pass the checks. the sub modules have wrong format string.

File: system/ubloxd/ublox_msg.cc [hunk 1]
```diff
@@ -390,7 +390,7 @@ kj::Array<capnp::word> UbloxMsgParser::parse_glonass_ephemeris(ubx_t::rxm_sfrbx_
     eph.setP4(data->p4());
     eph.setSvURA(glonass_URA_lookup.at(data->f_t()));
     if (msg->sv_id() != data->n()) {
-      LOGE("SV_ID != SLOT_NUMBER: %d %d", msg->sv_id(), data->n());
+      LOGE("SV_ID != SLOT_NUMBER: %d %" PRIu64, msg->sv_id(), data->n());
```
Comment (adeebshihadeh) [inline system/ubloxd/ublox_msg.cc:393]: why not llu?

**Review by adeebshihadeh — COMMENTED** [review]
can you open the matching rednose PR?

**Comment by deanlee** [comment]
> can you open the matching rednose PR?

opened at : https://github.com/commaai/rednose/pull/31

File: system/ubloxd/ublox_msg.cc (hunk 1 above)
Comment (deanlee) [reply system/ubloxd/ublox_msg.cc:393]: use llu will report error on my laptop.  the uint64_t is not the same as unsigned long long.  https://stackoverflow.com/questions/32198368/unsigned-long-long-conflict-with-uint64-t

the PRIu64 can correctly format uint64_t as lu or llu on different platforms

Comment (deanlee) [reply system/ubloxd/ublox_msg.cc:393]: another way is use llu, and convert uint64_t to (unsigned long long):

LOGE("SV_ID != SLOT_NUMBER: %d %llu", msg->sv_id(), (unsigned long long)data->n());

**Pull request merged**